shop/src/main/java/ro/msg/learning/shop/strategy/MostAbundantLocation.java

```diff
@@ -0,0 +1,90 @@
+package ro.msg.learning.shop.strategy;
+
+import lombok.AllArgsConstructor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import ro.msg.learning.shop.domain.Order;
+import ro.msg.learning.shop.domain.OrderDetail;
+import ro.msg.learning.shop.domain.Stock;
+import ro.msg.learning.shop.domain.key.OrderDetailId;
+import ro.msg.learning.shop.domain.key.StockId;
+import ro.msg.learning.shop.dto.CreateOrderDto;
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.exception.BadRequestException;
+import ro.msg.learning.shop.mapper.OrderMapper;
+import ro.msg.learning.shop.repository.OrderRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static ro.msg.learning.shop.message.Messages.BAD_PRODUCT;
+import static ro.msg.learning.shop.message.Messages.NEGATIVE_QUANTITY_EXCEPTION;
+
+@AllArgsConstructor
+@ConditionalOnProperty(name = "${strategy}", havingValue = "abundant")
+public class MostAbundantLocation implements OrderLocationStrategy {
+
+    private final StockRepository stockRepository;
+    private final ProductRepository productRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderDto create(CreateOrderDto createOrderDto) throws BadRequestException {
+
+        for (int i = 0; i < createOrderDto.getOrderProductDtoList().size(); i++) {
+            if (createOrderDto.getOrderProductDtoList().get(i).getQuantity() < 0) {
+                throw new BadRequestException(NEGATIVE_QUANTITY_EXCEPTION);
+            }
+        }
+        List<OrderDetail> orderDetails = new ArrayList<>();
+        OrderDto orderDto = OrderDto.builder()
+                .date(LocalDate.now())
+                .addressCity(createOrderDto.getAddressCity())
+                .addressCountry(createOrderDto.getAddressCountry())
+                .adressStreet(createOrderDto.getAddressStreet())
+                .adressCounty(createOrderDto.getAddressCounty())
+                .customerId(createOrderDto.getCustomerId())
+                .build();
+        Order order = orderMapper.toEntity(orderDto);
+
+        for (int i = 0; i < createOrderDto.getOrderProductDtoList().size(); i++) {
+            if (productRepository.findById(createOrderDto.getOrderProductDtoList().get(i).getProductId()).isEmpty()) {
+                throw new BadRequestException(BAD_PRODUCT);
+            }
+
+            List<UUID> orderedProductLocations = stockRepository.findSuitableLocation(createOrderDto.getOrderProductDtoList().get(i).getProductId(), createOrderDto.getOrderProductDtoList().get(i).getQuantity());
```

Comment:
I don't like this approach with haveing a `for (int i = 0; i < ....size(); i++)` and then using it like this `createOrderDto.getOrderProductDtoList().get(i)`.
I would like you to refactor some of this using for statements like this:
`for (OrderProductDto orderProductDto : orderProductDtoList)`